examples/plot_partial_dependence.py

```diff
@@ -0,0 +1,473 @@
+# number of bike rentals is decreasing when the wind speed is increasing for both
+# models. We also observe that :class:`~sklearn.neural_network.MLPRegressor` has much
+# smoother predictions than :class:`~sklearn.ensemble.HistGradientBoostingRegressor`.
+#
+# Now, we will look at the partial dependence plots for the categorical features.
+#
+# We observe that the spring season is the lowest bar for the season feature. With the
+# weather feature, the rain category is the lowest bar. Regarding the hour feature,
+# we see two peaks around the 7 am and 6 pm. These findings are in line with the
+# the observations we made earlier on the dataset.
+#
+# However, it is worth noting that we are creating potential meaningless
+# synthetic samples if features are correlated.
+#
+# .. _ice-vs-pdp:
+#
+# ICE vs. PDP
+# ~~~~~~~~~~~
+#
+# PDP is an average of the marginal effects of the features. We are averaging the
+# response of all samples of the provided set. Thus, some effects could be hidden. In
+# this regard, it is possible to plot each individual response. This representation is
+# called the Individual Effect Plot (ICE). In the plot below, we plot 50 randomly
+# selected ICEs for the temperature and humidity features.
+print("Computing partial dependence plots and individual conditional expectation...")
+tic = time()
+fig, axs = plt.subplots(ncols=2, figsize=(6, 4), sharey=True, constrained_layout=True)
+
+features_info = {
+    "features": ["temp", "humidity"],
+    # "centered": True,
+}
+
+pdp = PartialDependencePlot(
+    hgbdt_model,
+    **features_info,
+    **common_params,
+)
+# plot the figure
+pdp.fit_importance(X=X_train)
+for i, ax in enumerate(axs.ravel()):
+    _ = pdp.plot(feature_id=i, ax=ax, X=X_train)
+print(f"done in {time() - tic:.3f}s")
+_ = fig.suptitle("ICE and PDP representations", fontsize=16)
+
+# %%
+# We see that the ICE for the temperature feature gives us some additional information:
+# Some of the ICE lines are flat while some others show a decrease of the dependence
+# for temperature above 35 degrees Celsius. We observe a similar pattern for the
+# humidity feature: some of the ICEs lines show a sharp decrease when the humidity is
+# above 80%.
+#
+# Not all ICE lines are parallel, this indicates that the model finds
+# interactions between features. We can repeat the experiment by constraining the
+# gradient boosting model to not use any interactions between features using the
+# parameter `interaction_cst`:
+from sklearn.base import clone
+
+interaction_cst = [[i] for i in range(X_train.shape[1])]
+hgbdt_model_without_interactions = (
+    clone(hgbdt_model)
+    .set_params(histgradientboostingregressor__interaction_cst=interaction_cst)
+    .fit(X_train, y_train)
+)
+print(f"Test R2 score: {hgbdt_model_without_interactions.score(X_test, y_test):.2f}")
+
+# %%
+fig, axs = plt.subplots(ncols=2, figsize=(6, 4), sharey=True, constrained_layout=True)
```

Comment:
The x-label doesn't appear in the gallery, which makes it harder to follow the example. 
It seems that there's a line in PDP.plot() that should set it, though. Maybe it is the  `constrained_layout` that masks it?

Reply:
No, I forgot to add it.